### DISCUSSION TITLE
Add Straico as a provider

### DISCUSSION THREAD

**Discussion by RoboRiley** [cline/cline#551] [Feature Requests]
Straico is a popular and growing provider of LLM completions. They include Claude Sonnet 3.5, GPT o1-preview, and many other models.  Their API is pretty straightforward:
https://documenter.getpostman.com/view/5900072/2s9YyzddrR
I'd recommend the v1 API endpoints if possible.
I have run thousands of requests through their APIs and found them to be rock solid.

**Comment by meb427** [comment]
If they are compatible with openAI why not using the generic provider then?
